train.py

```diff
@@ -27,7 +27,7 @@
 WAVENET_PARAMS = './wavenet_params.json'
 STARTED_DATESTRING = "{0:%Y-%m-%dT%H-%M-%S}".format(datetime.now())
 SAMPLE_SIZE = 100000
-
+L2_REGULARIZATION_STRENGTH=0.
```

Comment:
The easy way is to define everything using flags and argparse, and then supply besides the flags a list of JSON files. The JSON files will be sequentially read and merged with the current set of options, and finally the dictionary from argparse will be merged with the current set, overriding everything. I can try and make a PR with this. I don't know of any libraries that may make this easier. The question then is whether the flags are global or we allow nesting via a magic nesting character in the flag name like `/` or `.`.

We could also define flags at different places (each file defines the global flag it needs), which works well with large codebases, but maybe the repo is not large enough for this.